src/mpi/coll/transports/gentran/gentran_impl.c

```diff
@@ -10,7 +10,7 @@
     - name        : MPIR_CVAR_PROGRESS_MAX_COLLS
       category    : COLLECTIVE
       type        : int
-      default     : 8
+      default     : 0
```

Comment:
I'm kind of surprised this claim. Even for small message sizes?

Reply:
This CVAR is only for Gentran. For small message size, the overhead for progressing each concurrent collective might be even worse relatively (progressing a graph schedule is relatively expensive when traversing graphs).

I think it is hard to find a good default for all the cases, however based on our performance profiling, default 0 in which case only progress one collective a time works the best for the OSU benchmarks.

Reply:
Are you using upstream OSU benchmarks? From what I gather, those benchmarks only issue 1 collective at a time. Each one is completed with `MPI_Wait` before the next issue starts.

Reply:
In typical scenarios, there is only one collective a time. If the collective algorithm only creates one schedule, then the value of this CVAR does not matter.
I think when Surabhi and I looked at this, it was a composition-based pipeline algorithm with both netmod and posix algorithms, in which case the gentran scheduler is consisted of one sub-schedule for each chunk. All subschedules even though they are sub-schedulers, they are actually progressed individually by progress engine. I think this change fixed a performance issue when the progress engine has too much overhead progressing too many schedules. In any case, do you know why this CVAR initially has a value "8"? 

The original code also may run into an issue that it always progress the first 8 schedules and return. When the first 8 schedules are stuck for some reason, program may hang. So this PR change it so that it makes sure each progress call actually makes progress. So potentially each progress call can process more schedulers (not just 8) because some schedules may not make any progress. This also motivated decreasing the value of this PR. 
In our profiling, processing only one schedule works best for most of cases. However, I can think of examples you would want to increase this CVAR, for example, in scenarios when one collective may not saturate the network.

Reply:
Sorry I confused this PR with something else. Please ignore the comments above. 
I fixed the commit messages and the PR description. Sorry for the confusion.

This commits actually disable the processing limit (by setting the CVAR to 0). So the progress engine will progress all pending collectives.